src/docs/message-handling.dox

```diff
@@ -0,0 +1,106 @@
+application-specific logging function that added a timestamp and logged
+the message to a file, and error messages might additionally be output to
+the \c stderr file stream.
+
+Additionally, applications will normally handle \c WT_PANIC as a special
+case. WiredTiger will always call the error handler callback with
+\c WT_PANIC in the case of a fatal error requiring database restart,
+however, WiredTiger cannot guarantee applications will see an application
+thread return \c WT_PANIC from a WiredTiger API call. For this reason, a
+correctly-written WiredTiger application will likely specify at least an
+error handler which will immediately exit or otherwise handle fatal errors.
+Note that no further WiredTiger calls are required after an error handler
+is called with \c WT_PANIC (and further calls will themselves immediately
+fail).
+
+The following is a programmatic example of creating a custom event handler:
+
+@snippet ex_event_handler.c Function event_handler
+
+The following is a programmatic example of configuring the previously defined custom event handler:
+
+@snippet ex_event_handler.c Configure event_handler
+
+@section message_handling_output Output format of the WT_EVENT_HANDLER
+
+The messages generated by the WT_EVENT_HANDLER can either be encoded in a flat string style format
+or JSON format.
+
+@subsection message_handling_output_flat_string Flat string format
+
+By default, all the messages going through the WT_EVENT_HANDLER are generated as flat strings. The
+example below shows a message following the flat string format:
+
+@code
+[1637732238:116292][11175:0x123456789012], test_salvage.test_salvage.test_salvage_api(column-fix), file:test_salvage.a.wt, WT_SESSION.salvage: [WT_VERB_SALVAGE_PROGRESS][INFO]: WT_SESSION.salvage
+@endcode
+
+@subsection message_handling_output_json JSON format
+
+When configuring a WiredTiger connection through ::wiredtiger_open or WT_CONNECTION::reconfigure, it
+is possible to configure the WT_EVENT_HANDLER interface to produce JSON encoded messages via the
+\c json_output configuration option. The \c json_output option takes a list message categories,
+where each category represents a method defined by the WT_EVENT_HANDLER interface, i.e. 'message'
+represents the ::handle_message interface. For instance, to print all \b messages and \b error
+events in JSON format, the following setting needs to be added to the connection configuration:
+
+@code
+json_output=(message, error)
+@endcode
+
+See ::wiredtiger_open for the full list of options that can be assigned to \c json_output. Any event
+handler interface that is not specified under the \c json_output setting will default to using the
+flat string format.
+
+The JSON output follows a specific schema with the following fields:
+
+<table>
+@hrow{Field, Description, Type}
+@row{category, Message category defined by \c WT_VERBOSE_CATEGORY, String}
+@row{category_id, Value representation of the message category, Int}
```

Comment:
small np: Maybe expand `Int` -> `Integer` for clarity.  
(sorry for the confusion, I realize my original comment said `int`)

Reply:
Fixed in 93511607b63da5a0d36e3043bc0d6ce7a9c5b35c